### PR TITLE
fix(angular.copy): support copying XML nodes

### LINKED ISSUE #5429: 1.2.5 angular-mocks copy change breaks tests using XML datatype
Apparently the XML datatype in Chrome/Firefox (and others?) contains pointers to itself.  The change here (https://github.com/angular/angular.js/commit/f69dc16241c8b631123ad0b09674f0a5e0ff32fe) copies the HTTP response.

If you are mocking an HTTP service with XML datatype as the return type, copy() never terminates.

### PR DESCRIPTION
**Pull request by petebacondarwin**
Closes #5429

### CODE REVIEW

**Comment by petebacondarwin** [comment]
Good old Internet Explorer:

```
IE 11.0.0 (Windows 8.1) angular copy should support XML nodes FAILED
	Expected '<?XML:NAMESPACE PREFIX = "PUBLIC" NS = "URN:COMPONENT" /><foo></foo>' to equal '<foo></foo>'.
```

**Comment by gkalpak** [comment]
Did you force-push a fix for the IE issue ?

File: test/AngularSpec.js [hunk 1]
```diff
@@ -386,6 +386,13 @@ describe('angular', function() {
       expect(aCopy).toBe(aCopy.self);
     });
 
+    it("should support XML nodes", function() {
```
Comment (gkalpak) [inline test/AngularSpec.js]: <sub>`"`  -->  `'` ? :blush: </sub>

File: src/Angular.js [hunk 2]
```diff
@@ -825,6 +825,8 @@ function copy(source, destination, stackSource, stackDest) {
       } else if (isRegExp(source)) {
         destination = new RegExp(source.source, source.toString().match(/[^\/]*$/)[0]);
         destination.lastIndex = source.lastIndex;
+      } else if (source.cloneNode) {
```
Comment (lgalfaso) [inline src/Angular.js]: should this be `} else if (isFunction(source.cloneNode)) {` ?

File: src/Angular.js [hunk 3]
```diff
@@ -825,6 +825,8 @@ function copy(source, destination, stackSource, stackDest) {
...
+          destination = source.cloneNode();
```
Comment (jbedard) [inline src/Angular.js]: Should this be a deep clone?

Comment (petebacondarwin) [inline src/Angular.js]: I guess you are right. I will add a unit test.

File: src/Angular.js (hunk 2 above)
Comment (petebacondarwin) [inline src/Angular.js]: I guess that is tighter

File: test/AngularSpec.js (hunk 1 above)
Comment (petebacondarwin) [inline test/AngularSpec.js]: :-) cut and paste from the original commit

**Comment by petebacondarwin** [comment]
@gkalpak - yes I hid my shame with a forced push. It was not really IE's fault (well yes it probably was, but it was mine too). Can you spot the mistake in my test:

```js
    it("should support XML nodes", function() {
      var anElement = document.createElement("foo");
      var theCopy = anElement.cloneNode();
      expect(copy(anElement.outerHTML)).toEqual(theCopy.outerHTML);
      expect(copy(anElement)).not.toBe(anElement);
    });
```

**Comment by petebacondarwin** [comment]
@gkalpak and @jbedard I added a commit with your suggested changes. Thanks

**Comment by gkalpak** [comment]
@petebacondarwin 
> Can you spot the mistake in my test

You mean besides the `"..."` ? :smile:
Oh...and the misplaced parenthesis :)

LGTM

**Comment by petebacondarwin** [comment]
Thanks. I'll merge

**Pull request merged**